src/rgw/rgw_admin.cc

```diff
@@ -2865,11 +2865,13 @@ int main(int argc, const char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = strict_iecstrtoll(val.c_str(), &err);
-      if (!err.empty()) {
-        cerr << "ERROR: failed to parse max size: " << err << std::endl;
-        return EINVAL;
-      }
+      if (val.length() > 0 && val[0] != '-') {
+        max_size = strict_iecstrtoll(val.c_str(), &err);
```

Comment:
can you try `strict_iec_cast<long long>()` here instead? it should handle the negative